lib/client.js

```diff
@@ -702,16 +729,11 @@ export default (function INIT() {
                 return;
               };
 
-              // filter via .ftpignore
-              const ftpignore = self.getFilePath('.ftpignore');
-              const ignoreFilter = Ignore();
-              if (FS.existsSync(ftpignore)) {
-                ignoreFilter.add(FS.readFileSync(ftpignore).toString());
-              }
-
               // remove ignored locals
-              for (let i = locals.length - 1; i >= 0; i--) {
-                if (ignoreFilter.ignores(locals[i].name)) {
+              for (var i = locals.length - 1; i >= 0; i--) {
+                if (typeof self.ignoreFilter === "undefined") break;
+
+                if (self.ignoreFilter.ignores(locals[i].name)) {
```

Comment:
How can you be so sure that `self.ignoreFilter` is initialized at this point? As far as I see, you initialize `self.ignoreFilter` every time `self.client.checkIgnore(...)` is called.

When I try to syncLocalRemote before saving any file in the editor, `self.ignoreFilter` is never initialized and no file is ignored at all. So the ignore file has to be reloaded before the for-loop here.

You could split `checkIgnore` into `checkIgnore` and something like `updateIgnore`   which initializes `self.ignoreFilter`. Then you could call `updateIgnore` before the for loop here